contracts/EventBasedPredictionMarket.sol

```diff
@@ -0,0 +1,318 @@
+        require(shortToken.burnFrom(msg.sender, shortTokensToRedeem));
+
+        // expiryPercentLong is a number between 0 and 1e18. 0 means all collateral goes to short tokens and 1e18 means
+        // all collateral goes to the long token. Total collateral returned is the sum of payouts.
+        uint256 longCollateralRedeemed = (longTokensToRedeem * collateralPerPair * expiryPercentLong) / (1e18**2);
+        uint256 shortCollateralRedeemed = (shortTokensToRedeem * collateralPerPair * (1e18 - expiryPercentLong)) /
+            (1e18**2);
+
+        collateralReturned = longCollateralRedeemed + shortCollateralRedeemed;
+        collateralToken.safeTransfer(msg.sender, collateralReturned);
+
+        emit PositionSettled(msg.sender, collateralReturned, longTokensToRedeem, shortTokensToRedeem);
+    }
+
+    /****************************************
+     *          INTERNAL FUNCTIONS          *
+     ****************************************/
+
+    /**
+     * @notice Request a price in the optimistic oracle for a given request timestamp and ancillary data combo. Set the bonds
+     * accordingly to the deployer's parameters. Will revert if re-requesting for a previously requested combo.
+     */
+    function _requestOraclePrice() internal {
+        OptimisticOracleInterface optimisticOracle = getOptimisticOracle();
+
+        collateralToken.safeApprove(address(optimisticOracle), proposerReward);
+
+        optimisticOracle.requestPrice(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData,
+            collateralToken,
+            proposerReward
+        );
+
+        // Set the Optimistic oracle liveness for the price request.
+        optimisticOracle.setCustomLiveness(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData,
+            optimisticOracleLivenessTime
+        );
+
+        // Set the Optimistic oracle proposer bond for the price request.
+        optimisticOracle.setBond(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData,
+            optimisticOracleProposerBond
+        );
+
+        // Make the request an event-based request.
+        OptimisticOracleInterfaceEventBased(address(optimisticOracle)).setEventBased(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData
+        );
+
+        priceRequested = true;
+    }
+
+    /**
+     * @notice Returns a number between 0 and 1e18 to indicate how much collateral each long and short token are entitled
+     * to per collateralPerPair.
+     * @param _expiryPrice price from the optimistic oracle for the LSP price identifier.
+     * @return expiryPercentLong to indicate how much collateral should be sent between long and short tokens.
+     */
+    function percentageLongCollAtExpiry(int256 _expiryPrice) internal view returns (uint256) {
```

Comment:
So this works but is overly complicated in my opinion. This method would be better if we wanted to allow people to create binary options, use price identifiers such as ETHUSD and set different `strikePrices`. But since the goal here is to create a prediction market and strikePrice is defaulted to 1e18 anyways, I think it'd be better to:
- Remove `strikePrice`
- Remove this function `percentageLongCollAtExpiry`
- add basic validation within `getExpirationPrice()` that makes sure returned values are either 0, 1e18 or 0.5e18

Does this make sense and does it seem simpler to you?